docker/lwua-ingest/lwua-py/lwua/watcher.py

```diff
@@ -0,0 +1,84 @@
+import time
+from datetime import datetime
+import os
+from pathlib import Path
+from dotenv import load_dotenv
+from abc import ABC, abstractmethod
+import logging
+from lwua.helpers import enable_logging, resolve_path
+from lwua.graphdb import (
+    ingest_data_file,
+    delete_data_file,
+    get_registry_of_lastmod,
+)
+
+log = logging.getLogger(__name__)
+
+class FolderChangeObserver(ABC):
+    @abstractmethod
+    def added(self, fname: str, lastmod: datetime = None):
+        pass
+
+    @abstractmethod
+    def removed(self, fname: str):
+        pass
+
+    @abstractmethod
+    def changed(self, fname: str, lastmod: datetime = None):
+        pass
+
+class FolderChangeDetector:
+    def __init__(self, folder_to_inspect):
+        self.root = Path(folder_to_inspect)
+        while not self.root.exists():
+            log.info(f"Waiting for {self.root} to exist")
+            time.sleep(1)
+        log.info(f"Watching {self.root}")
+
+    def report_changes(self, known_lastmod_by_fname: dict = {}, observer: FolderChangeObserver = None):
+        current_lastmod_by_fname = {p: datetime.utcfromtimestamp(os.path.getmtime(p)) for p in self.root.glob('**/*') if p.is_file()}
+        log.info(f"current_lastmod_by_fname: {current_lastmod_by_fname}")
+        for fname in known_lastmod_by_fname:
+            if fname not in current_lastmod_by_fname:
+                observer.removed(fname)
+        for fname, lastmod in current_lastmod_by_fname.items():
+            if fname not in known_lastmod_by_fname:
+                log.info(f"new file {fname} with lastmod {lastmod}")
+                observer.added(fname, lastmod)
+            elif lastmod > known_lastmod_by_fname[fname]:
+                observer.changed(fname, lastmod)
+                
+        return current_lastmod_by_fname
+
+class IngestChangeObserver(FolderChangeObserver):
```

Comment:
this is not part of the generic watcher 

move this (together with the import of the lwua.graph dependency to either - ingest.py, separate observer.py (if the ingest is already getting too fat)